File: Programmers/Level1/최대공약수와최소공배수/최대공약수와최소공배수.js
function gcd(a, b) {
    while (b) {
        [a, b] = [b, a % b];
    }
    return a;
}

function lcm(a, b) {
    return parseInt((a * b) / gcd(a, b));
}

function solution(n, m) {
    return [gcd(n, m), lcm(n, m)];
    // return [gcd(n, m), parseInt((n * m) / gcd(n, m))];
}

// 다른 풀이
function solution(n, m) {
    const gcd = (a, b) => (!b ? a : gcd(b, a % b));
    const lcm = (a, b) => (a * b) / gcd(a, b);

    return [gcd(n, m), lcm(n, m)];
}

File: Programmers/Level1/최대공약수와최소공배수/최대공약수와최소공배수.py
def gcd(a, b):
    while b:
        a, b = b, a % b
    return a
    
def lcm(a, b):
    return a * b // gcd(a, b)
    
def solution(n, m):
    return [gcd(n, m), lcm(n, m)]
    # return [gcd(n, m), n * m // gcd(n, m)]

# lambda 함수 사용
def solution(n, m):
    gcd = lambda a, b : a if not b else gcd(b, a % b)
    lcm = lambda a, b : a * b // gcd(a, b)
    return [gcd(n, m), lcm(n, m)]